server/channels/store/sqlstore/scheduled_post_store.go

```diff
@@ -69,8 +69,8 @@ func (s *SqlScheduledPostStore) scheduledPostToSlice(scheduledPost *model.Schedu
 func (s *SqlScheduledPostStore) CreateScheduledPost(scheduledPost *model.ScheduledPost) (*model.ScheduledPost, error) {
 	scheduledPost.PreSave()
 	maxMessageSize := s.getMaxMessageSize()
-	if err := scheduledPost.IsValid(maxMessageSize); err != nil {
-		return nil, errors.Wrap(err, "failed to validate scheduled post")
+	if err := scheduledPost.IsMessageLengthValid(maxMessageSize); err != nil {
+		return nil, errors.Wrap(err, "failed to validate scheduled post message length")
```

Comment:
I wanted to comment on this regardless. Why are we doing validation checks on the store layer? We've already done some validation in the app layer and it's both more future proof and easier for reviewers to have all input validation in one place.

Reply:
This seems to be causing a lot of confusion for every reviewer. The reason it was this way was because the max allowable message length is available only in store layer, so I was checking for that in store layer and making all other checks in app layer. But this has caused some confusion for all reviewers so far.

So, what I've done now is to make a function to obtain the max message length and now I can perform all check in app layer, bringing all checks in one place.